### PR TITLE
Add GPT-OSS architecture and plumbing

### PR DESCRIPTION
**Pull request by gracee3**
## Summary
- split the mergeable GPT-OSS architecture and plumbing work out of #6
- rebase this slice onto current `main`
- keep the deferred structural refactor and broader runtime follow-ups out of this PR

## Included
- `gpt_oss.rs` architecture implementation and registration
- GPT-OSS config/model plumbing in engine, worker config, runner, and loader
- GPT-OSS MXFP4 expert kernel support
- quantization detection needed for GPT-OSS checkpoints

## Addressed From Review
- resolved the current `gpu_engine.rs` drift against `main`
- removed stale cleanup items called out in review
- dropped unrelated README churn from this branch

## Deferred To Follow-up
- the broader `gpu_layer.rs` structural refactor
- the `gpu_runner.rs` fusion rework
- additional GPT-OSS runtime and graph-decode follow-ups that do not need to block this mergeable plumbing slice

## Validation
- `cargo check -p rvllm-model-runner --features cuda`
- `cargo check -p rvllm-worker --features cuda`
- `cargo check -p rvllm-engine --features cuda`

### CODE REVIEW

**Comment by m0at** [comment]
ok somehow, someway, we are gonna try to get this PR back for you, lemme think about it tongiht haha

**Comment by gracee3** [comment]
Quick update: we’ve achieved a successful end-to-end final-token validation against a PyTorch oracle (transformer through logits) using our current harness/proof path.

We’re now transitioning off that slow/validated path and implementing the equivalent projection/runtime behavior directly in the Rust engine.

It probably makes sense to close this out, and I’d be happy to open a fresh PR once I’ve had a chance to get caught up with the latest work here.
